lib/util/async_store_listener.dart

```diff
@@ -5,21 +5,26 @@ import '../l10n/l10n_from_string.dart';
 import 'async_store.dart';
 import 'observer_consumers.dart';
 
-class AsyncStoreListener extends StatelessWidget {
-  final AsyncStore asyncStore;
+class AsyncStoreListener<T> extends StatelessWidget {
+  final AsyncStore<T> asyncStore;
   final String? successMessage;
+  final String Function(
+    BuildContext context,
+    AsyncStateData<T> asyncStore,
+  )? successMessageBuilder;
   final Widget child;
 
   const AsyncStoreListener({
     Key? key,
     required this.asyncStore,
     this.successMessage,
+    this.successMessageBuilder,
     required this.child,
   }) : super(key: key);
```

Comment:
Assert that `successMessage` NAND `successMessageBuilder`

Reply:
Actually, successMessage is never used anyways, and we will most likely always be using the builder, so I would just remove successMessage all together